service/src/main/java/org/cbioportal/genome_nexus/service/CachedExternalResourceFetcher.java

```diff
@@ -3,9 +3,10 @@
 import org.cbioportal.genome_nexus.service.exception.ResourceMappingException;
 
 import java.util.List;
+import java.util.Optional;
 
 public interface CachedExternalResourceFetcher<T>
 {
-    T fetchAndCache(String id) throws ResourceMappingException;
+    Optional<T> fetchAndCache(String id) throws ResourceMappingException;
```

Comment:
I still have mixed feelings about defining an `Optional` at the interface level.